### PR TITLE
Faster Vroom Vroom

### PR DESCRIPTION
**Pull request by duplexsystem**
updated some things and so build should be even faster now.

### CODE REVIEW

**Comment by duplexsystem** [comment]
Don't merge yet there are a few things I'm going to add later

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,13 +13,13 @@ jobs:
         java: [11, 8, 14]
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v2.2.0
```
Comment (mio-19) [inline .github/workflows/build.yml]:
```suggestion
    - uses: actions/checkout@v2
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -45,7 +45,7 @@ jobs:
       run: |
         ./yapfa full
     - name: Upload Artifact
-      uses: actions/upload-artifact@v2-preview
+      uses: actions/upload-artifact@v2.0.1
```
Comment (mio-19) [inline .github/workflows/build.yml]:
```suggestion
      uses: actions/upload-artifact@v2
```

Comment (duplexsystem) [reply .github/workflows/build.yml]: Why?

File: .github/workflows/build.yml (hunk 1 above)
Comment (duplexsystem) [reply .github/workflows/build.yml]: Why?

**Comment by duplexsystem** [comment]
@tr7zw PR is now ready for review. It includes:
About 1 minute faster build times
No longer builds on .MD changes
Updated all actions
Cancels old builds
Uploads latest builds to release tab and deletes old releases.
No longer builds 2 twice when doing a PR from a local branch

**Review by tr7zw — APPROVED** [review]

**Pull request merged**